Add RemoveAccountModal to replace default-styled alert dialog

dguenther opened the pull request:
Adds a RemoveAccountModal to replace the default alert styling.

## Account has assets

<img width="916" alt="Screenshot 2024-01-17 185031" src="https://github.com/iron-fish/ironfish-node-app/assets/767083/f1ca7ab6-1e89-49dc-a6c0-b276d5e7917c">

## Account is syncing

<img width="922" alt="Screenshot 2024-01-18 112205" src="https://github.com/iron-fish/ironfish-node-app/assets/767083/020c9584-3a07-439a-8e0e-18d559867663">

## Account is synced and has no assets

<img width="916" alt="Screenshot 2024-01-17 185050" src="https://github.com/iron-fish/ironfish-node-app/assets/767083/da9a2d1b-0b38-42a7-8ca2-cd4f8207bad2">

## Entered account name incorrectly
<img width="916" alt="Screenshot 2024-01-17 185714" src="https://github.com/iron-fish/ironfish-node-app/assets/767083/3ed0f130-db95-4a17-94e8-35d84656c237">


Fixes IFL-2075

skylarrichard123 reviewed (changes requested):
looks good other than that you can delete an account without typing in the account name in the delete confirmation modal, bypassing the point of the modal haha

dguenther commented:
Fixed that issue and updated the screenshots to the latest designs. This is waiting for a follow-up on what should display in the "Syncing" state, but I took a stab at it.

dguenther commented:
Updated it with the new syncing state.
